src/plugins/content_management/server/event_stream/es/init/mappings.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { estypes } from '@elastic/elasticsearch';
+
+export const mappings: estypes.MappingTypeMapping = {
+  dynamic: false,
+  properties: {
+    /**
+     * Every document indexed to a data stream must contain a `@timestamp`
+     * field, mapped as a `date` or `date_nanos` field type.
+     */
+    '@timestamp': {
+      type: 'date',
+    },
+
+    /** Subject is the content item who/which performed the event. */
+    subjectType: {
```

Comment:
How are we going to keep garbage from being added to this index.  Someone typo's a `predicate` name, for instance?

For the event log, we require clients pre-register the types of docs they want to create, by sending in the value for two fields - one fixed (field must be this value), the other an array (field must be one of these values) - and if we see docs that don't match any registrations, we don't index it.

I think following that pattern you could have clients register `predicate`s they plan on using, and optionally `subjectType` / `objectType` values that are supported.  This also makes it easy to find what a plugin will end up generating, in a single place in setup, rather than searching through the rest of the code.  And lessens the chance of indexing garbage.

Reply:
> How are we going to keep garbage from being added to this index. Someone typo's a predicate name, for instance?

Good point. And we have it covered.

Currently, the Event Stream is not exposed externally, so only Content Management can write to it. So we know all event types and all payloads of those events.

However, I would like to have a registry of the events, similar to what you are suggesting. We have a ticket for that: https://github.com/elastic/kibana/issues/153259 Now, I am maybe the only champion of implementing that registry in our team, so maybe we will not implement the registry in the end. (Which should be OK, as long as Event Stream stays internal to Content Management.)

But, we might also want to expose some events through HTTP. Like a "view" or "open" event could be logged straight from the browser. For that we have a ticket as well: https://github.com/elastic/kibana/issues/153261 And the we will not expose the ability to log any event, we will only allow to log events through HTTP which were specifically allowlisted and validated (from the issue):

> Only allow to emit events from RPC, which are explicitly in the allow list of RPC events.